packages/integrations/node/src/nodeMiddleware.ts

```diff
@@ -1,17 +1,27 @@
 import type { NodeApp } from 'astro/app/node';
-import type { IncomingMessage, ServerResponse } from 'node:http';
+import type { ServerResponse } from 'node:http';
 import type { Readable } from 'stream';
 import { createOutgoingHttpHeaders } from './createOutgoingHttpHeaders';
 import { responseIterator } from './response-iterator';
-import type { Options } from './types';
+import type { ErrorHandlerParams, Options, RequestHandlerParams } from './types';
 
 export default function (app: NodeApp, mode: Options['mode']) {
```

Comment:
```suggestion
// Disable no-unused-vars to avoid breaking signature change
// eslint-disable-next-line @typescript-eslint/no-unused-vars
export default function (app: NodeApp, _mode: Options['mode']) {
```